src/routes/synapse/room_events.rs

```diff
@@ -186,24 +188,73 @@ async fn process_room_event(
         friendships_repository: &repos.friendships,
         friendship_history_repository: &repos.friendship_history,
     };
+
+    // The only case where we don't create the friendship if it didn't exist
+    // If they are still no friends, it's unnecessary to create a friendship
+    if current_status == new_status {
+        return Ok(RoomEventResponse {
+            event_id: room_id.to_string(),
+        });
+    }
+
+    // Start transaction
+    let transaction = friendship_ports.db.start_transaction().await;
+    if transaction.is_err() {
+        let err = transaction.err().unwrap();
+        log::error!("Couldn't start transaction to store friendship update {err}");
+        return Err(SynapseError::CommonError(CommonError::Unknown));
+    }
+
     // UPDATE FRIENDSHIP ACCORDINGLY IN DB
-    update_friendship_status(
+    let transaction = update_friendship_status(
         &friendship,
         acting_user,
         &second_user,
-        current_status,
         new_status,
         room_info,
         friendship_ports,
+        transaction,
     )
     .await?;
 
+    // If it's a friendship request event and the request contains a message, we send a message event to the given room.
+    if room_event == FriendshipEvent::REQUEST {
+        if let Some(val) = room_message_body {
+            let mut retry_count = 0;
+            loop {
```

Comment:
can you move this to a separate function in this module that is called something like `store message in synapse room`? So that this function is more easy to follow